static/js/components/Spectrum.jsx

```diff
@@ -0,0 +1,243 @@
+      )}
+      {spectrum.observers.length > 0 && (
+        <div>
+          <Typography variant="h6">Observed by</Typography>
+          {spectrum.observers.map((observer) => (
+            <UserContactInfo user={observer} key={observer.id} />
+          ))}
+        </div>
+      )}
+      <Plot
+        className={classes.plot}
+        url={`/api/internal/plot/spectroscopy/${spectrum.obj_id}?spectrumID=${spectrum.id}`}
+      />
+      {spectrum.owner_id === uid && (
+        <Button
+          onClick={() => {
+            setOpen(true);
+          }}
+          data-testid="delete-spectrum-button"
+        >
+          Delete Spectrum
+        </Button>
+      )}
+      <DownloadLink
+        filename={filename}
+        exportFile={() => data}
+        style={{ color: "rgba(0, 0, 0, 0.87)", fontWeight: 500 }}
+        label="Download ASCII Spectrum"
+        tagName={Button}
+      />
+      <Dialog
+        open={open}
+        aria-labelledby="simple-modal-title"
+        aria-describedby="simple-modal-description"
+        onClose={() => {
+          setOpen(false);
+        }}
+        className={classes.detailedSpecButton}
+      >
+        <DialogContent>
+          <div>
+            <Typography variant="h6">
+              Are you sure you want to do this?
+            </Typography>
+            The following operation <em>permanently</em> deletes the spectrum
+            from the database. This operation cannot be undone and your data
+            cannot be recovered after the fact. You will have to upload the
+            spectrum again from scratch.
+          </div>
+          <div>
+            <Button
+              onClick={() => {
+                setOpen(false);
+              }}
+            >
+              No, do not delete the spectrum.
+            </Button>
+            <Button
+              onClick={async () => {
+                const result = await dispatch(deleteSpectrum(spectrum.id));
+                if (result.status === "success") {
+                  dispatch(showNotification("Spectrum deleted."));
+                }
+              }}
+              data-testid="yes-delete"
+            >
+              Yes, delete the spectrum.
+            </Button>
+          </div>
+        </DialogContent>
+      </Dialog>
+    </div>
+  );
+};
+
+const user = {
+  id: PropTypes.number.isRequired,
+  contact_email: PropTypes.string.isRequired,
+  first_name: PropTypes.string,
+  last_name: PropTypes.string,
+  username: PropTypes.string.isRequired,
+};
+
+DetailedSpectrumView.propTypes = {
+  spectrum: PropTypes.shape({
+    id: PropTypes.string,
+    obj_id: PropTypes.number,
+    owner_id: PropTypes.number,
+    instrument_id: PropTypes.number,
+    reducers: PropTypes.arrayOf(user),
+    observers: PropTypes.arrayOf(user),
+    owner: PropTypes.shape(user),
+    original_file_string: PropTypes.string,
+    original_file_filename: PropTypes.string,
+  }).isRequired,
+};
+
+const SpectrumPage = ({ route }) => {
```

Comment:
```suggestion
const SpectraPage = ({ route }) => {
```
This renders multiple spectra, right? Alternatively, `SourceSpectra`/`ObjSpectra`

Reply:
Same comment for file name

Reply:
done